(SERVER-2804) Add release notes for 6.12.0 (#2356)
puppetlabs/puppetserver: documentation/release_notes.markdown

[puppetserver.conf]: ./config_file_puppetserver.markdown
[product.conf]: ./config_file_product.markdown

## Puppet Server 6.12.0

Released 3 June 2020

> ### Resolved Issue

Author: lmk if we're comfortable calling the jruby issue "resolved"..

> ### Resolved Issue

Reviewer: Minor, but lower case for issue, so it's consistent.

- JRuby has been bumped to 9.2.11.1 again, with `invokedynamic.yield` set to false to resolve a stackoverflow error. [SERVER-2793](https://tickets.puppetlabs.com/browse/SERVER-2793)

> ### Deprecation

Author: Wasn't sure which heading to use here, lmk if there's something else we've used in the past.

Reviewer 2: "Deprecation" seems fine to me.

> - We have deprecated the v1 metrics endpoint, which was recently disabled by default. It will be removed entirely in a future release. Please use the v2 endpoint instead. [TK-486](https://tickets.puppetlabs.com/browse/TK-486)

Reviewer: ```suggestion
- The v1 metrics endpoint, which was recently disabled by default, is now deprecated. Instead, use the v2 endpoint. [TK-486](https://tickets.puppetlabs.com/browse/TK-486)
```

Reviewer: Usually we avoid saying things about future releases in the docs, and being deprecated implies that it will be removed later anyway I think.

Author: Good point, thanks! updated 😊
